### ISSUE TITLE
Default config (emoji.yaml) fails to converge

### ISSUE THREAD

**Issue by evancasey** [preddy5/Im2Vec#13]
Hey, thanks so much for the code release. Really excited about the potential of this project.

I think the default config might have some incorrect hyper parameters? After 300-400 epochs I'm still not seeing the model able to reconstruct the emoji dataset.

Any guidance on what parameters to use would be appreciated. I tried what was suggested in  https://github.com/preddy5/Im2Vec/issues/5 but did not see an improvement.

![recons_VectorVAEnLayers_0383](https://user-images.githubusercontent.com/1900535/133013854-4baca034-fb3f-4ec9-b9b6-92c73f8d2e04.png)

**Comment by preddy5** [comment]
Hey @evancasey 
This tends to happen with high learning rates, could you try decreasing the learning rate https://github.com/preddy5/Im2Vec/blob/master/configs/emoji.yaml#L23 to 0.0001 and see if the changes anything.

**Comment by evancasey** [comment]
I'm starting to see better results after reducing LR to 0.0001, however it takes much longer to train than your logged results. Would you suggest tuning LR some more or are there any other hyperparameters that come to mind?

**Comment by evancasey** [comment]
![recons_VectorVAEnLayers_0547](https://user-images.githubusercontent.com/1900535/134449333-bf619350-51c0-4a58-9aab-533e94cb788b.png)
After 500 epochs, LR=0.0001

**Comment by evancasey** [comment]
After 2000 epochs it still does not converge

**Comment by preddy5** [comment]
Hey @evancasey I think you might have to tune the LR a bit, could you try 0.0002 maybe, also increasing https://github.com/preddy5/Im2Vec/blob/master/configs/emoji.yaml#L8 the radius helps with the initial part of the optimization maybe try 20. Hopefully, this helps with the optimization.

**Comment by Emasoft** [comment]
Same here. Tried everything. Almost 3000 epochs and it still does not converge. 
There are 3 logical possibilities:
1 - The code in this repository is not the one used in the paper (i.e. notebooks?)
2 - There is a secret way to tune those parameters with a separate tool/script
3 - The results in the paper are made up
Excluding the third, both the 1st and the 2nd explanations are easy to solve. 
You just need to merge/add the source code of the notebook or of the tool with this repo.
Are you still under deadlines? I hope you will find a moment to fix this. This is an important work.
Thanks. 👍

**Comment by preddy5** [comment]
Hey @Emasoft 
There are many reasons why the network might not have converged when you train it. 
The repo includes an example training run please use that for debugging your training. For each new training run, the code also creates a copy of the training code in the log directory, so the example training run also contains a backup of the code used for that specific training and trained weights, you can use them for debugging as well. 

Hope that helps.
